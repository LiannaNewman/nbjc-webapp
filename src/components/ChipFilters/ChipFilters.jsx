import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { withStyles } from '@material-ui/core/styles';
import { Chip, Typography } from '@material-ui/core';
import CheckOutlinedIcon from '@material-ui/icons/CheckOutlined';

const styles = () => ({
  chip: {
    margin: 10,
  },
  wrapper: {
    display: 'block',
  },
});

const ChipFilters = ({
  chips, onChipSelected, classes, chipSize, overrideClasses,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  return (
    <div className={cx(classes.wrapper, overrideClasses.wrapper)}>
      {chips.map((chip, i) => (
        <Chip
          label={
            <Typography variant="body2">{chip.name}</Typography>
          }
          key={chip.name.replace(/\w/, '')}
          variant={chip.isSelected ? 'default' : 'outlined'}
          size={chipSize || matches ? 'medium' : 'small'}
          onClick={() => onChipSelected(i)}
          icon={chip.isSelected ? <CheckOutlinedIcon /> : null}
          color="primary"
          className={cx(classes.chip, overrideClasses.chip)}
        />
      ))}
    </div>
  );
};

ChipFilters.propTypes = {
  chips: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    isSelected: PropTypes.bool,
  })).isRequired,
  onChipSelected: PropTypes.func.isRequired,
  classes: PropTypes.shape({}).isRequired,
  chipSize: PropTypes.oneOf(['small', 'medium', 'large']),
  overrideClasses: PropTypes.shape({}),
};

ChipFilters.defaultProps = {
  chipSize: null,
  overrideClasses: {},
};

export default withStyles(styles)(ChipFilters);
