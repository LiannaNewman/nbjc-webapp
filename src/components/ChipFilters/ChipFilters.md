## ChipFilters

```jsx
const chips = [{
    name: 'Black Friendly',
  }, {
    name: 'Inclusive',
  }, {
    name: 'Black Owned',
  }, {
    name: 'Gender Neutral Restrooms',
  }, {
    name: 'Accessible',
  }, {
    name: 'Queer hangout space',
  }, {
    name: 'Trans friendly',
  }, {
    name: 'Queer owned',
}];
const [activeChips, setActiveChips] = React.useState(chips);
const onChipSelected = (i) => {
  const chips = [...activeChips];
  chips[i].isSelected = !chips[i].isSelected;
  setActiveChips(chips);
}
<ChipFilters chips={activeChips} onChipSelected={onChipSelected} />
```