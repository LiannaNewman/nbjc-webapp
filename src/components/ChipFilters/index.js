import ChipFilters from './ChipFilters';

export default ChipFilters;
