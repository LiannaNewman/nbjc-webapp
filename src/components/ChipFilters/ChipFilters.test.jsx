import React from 'react';
import { render } from '@testing-library/react';

import ChipFilters from './ChipFilters';

describe('ChipFilters', () => {
  test('render', () => {
    const chips = [{
      name: 'Black Friendly',
    }, {
      name: 'Inclusive',
    }, {
      name: 'Black Owned',
    }, {
      name: 'Gender Neutral Restrooms',
    }, {
      name: 'Accessible',
    }, {
      name: 'Queer hangout space',
    }, {
      name: 'Trans friendly',
    }, {
      name: 'Queer owned',
    }];
    const { asFragment } = render(<ChipFilters chips={chips} onChipSelected={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
