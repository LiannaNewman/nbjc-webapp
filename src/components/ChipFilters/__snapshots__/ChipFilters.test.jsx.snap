// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipFilters render 1`] = `
<DocumentFragment>
  <div
    class="ChipFilters-wrapper-2"
  >
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Black Friendly
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Inclusive
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Black Owned
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Gender Neutral Restrooms
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Accessible
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Queer hangout space
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Trans friendly
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-1 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Queer owned
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
